bike_kollective/lib/data/provider/active_user.dart

```diff
@@ -1,83 +1,58 @@
 import 'package:bike_kollective/data/model/user.dart';
 import 'package:bike_kollective/data/provider/database.dart';
+import 'package:firebase_auth/firebase_auth.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:google_sign_in/google_sign_in.dart';
 
 // Provides updates about the user account that is logged in (if any)
 final activeUserProvider = StateNotifierProvider<ActiveUserNotifier, UserModel?>((ref) {
   final dbAccess = ref.watch(databaseProvider);
   return ActiveUserNotifier(dbAccess);
 });
 
-// The logged in user is determined by this class
+// The logged-in user is determined by this class
 class ActiveUserNotifier extends StateNotifier<UserModel?> {
   final BKDB dbAccess;
+  final GoogleSignIn _googleSignIn = GoogleSignIn(
+    scopes: [
+      'email',
+      'https://www.googleapis.com/auth/userinfo.profile',
+    ],
+  );
+
   ActiveUserNotifier(this.dbAccess) : super(null);
 
   void signOut() {
     // Log out of the current user's account
+    _googleSignIn.signOut();
     state = null;
   }
 
-  void signIn(String uid) {
-    // Respond to successful login with the given authentication UID
-    dbAccess.getUserByUid(uid)
-    .then((user) {
-      state = user;
-    })
-    .catchError((error) {
-      // TODO - send sign in error notification to error notifier?
+  Future<void> signIn() async {
+    try {
+      // Initiate the Google sign-in flow
+      final GoogleSignInAccount? googleUser = await _googleSignIn.signIn();
+      if (googleUser == null) {
+        // The user canceled the sign-in
+        state = null;
+        return;
+      }
+
+      final GoogleSignInAuthentication googleAuth = await googleUser.authentication;
+      final String? idToken = googleAuth.idToken;
+
+      if (idToken != null) {
+        // Fetch user from the database or use dummy data for testing
+        final user = await dbAccess.getUserByUid(googleUser.id);
+        
+        // If user is null, create a dummy user for testing
+        state = user ?? UserModel.newUser(uid: googleUser.id);
```

Comment:
We should put a todo to replace this with correct handling for database errors. But this is fine for now.